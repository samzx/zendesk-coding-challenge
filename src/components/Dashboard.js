import React from "react";
import ReactDOM from "react-dom";
import Modal from "react-modal";

import Header from "./Header";
import Drawer from "./Drawer";
import Ticket from "./Ticket";

const customStyles = {
  content: {
    maxWidth: "720px",
    margin: "0 auto",
    overflow: "inherit",
    top: "0",
    left: "0",
    right: "0",
    bottom: "0",
    padding: "0"
  }
};

class Dashboard extends React.Component {
  state = {
    showModal: false,
    data: null,
    loading: false,
    errors: null,
    currentTicket: null
  };

  setTicket = id => {
    this.setState({ currentTicket: id, showModal: true });
  };

  setData = data => {
    this.setState({ data });
  };

  render() {
    const { data } = this.state;
    return (
      <div className="dashboard">
        <Drawer
          data={data}
          setData={this.setData}
          currentTicket={this.state.currentTicket}
          setTicket={this.setTicket}
          fetchListings={this.fetchListings}
        />
        <Modal
          ariaHideApp={false}
          id="the-react-modal"
          isOpen={this.state.showModal}
          shouldCloseOnOverlayClick={true}
          style={customStyles}
          onRequestClose={() => {
            this.setState({ showModal: false });
          }}
        >
          <div className="modal-header">
            <span
              className="modal-close"
              onClick={() => this.setState({ showModal: false })}
            >
              {"ğŸ”™"}
            </span>
          </div>
          <div className="banner" />
          <Ticket currentTicket={this.state.currentTicket} data={data} />
        </Modal>
      </div>
    );
  }
}

export default Dashboard;
