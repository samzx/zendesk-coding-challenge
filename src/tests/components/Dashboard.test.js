import React from "react";
import { shallow } from "enzyme";
import Dashboard from "../../components/Dashboard";

describe("<Dashboard />", () => {
  it("should render <Dashboard />", () => {
    const wrapper = shallow(<Dashboard />);
    expect(wrapper).toMatchSnapshot();
  });
});
