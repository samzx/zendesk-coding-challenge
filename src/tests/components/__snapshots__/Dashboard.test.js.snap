// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dashboard correctly 1`] = `
<div
  className="dashboard"
>
  <Drawer
    currentTicket={null}
    data={null}
    setData={[Function]}
    setTicket={[Function]}
  />
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    id="the-react-modal"
    isOpen={false}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "bottom": "0",
          "left": "0",
          "margin": "0 auto",
          "maxWidth": "720px",
          "overflow": "inherit",
          "padding": "0",
          "right": "0",
          "top": "0",
        },
      }
    }
  >
    <div
      className="modal-header"
    >
      <span
        className="modal-close"
        onClick={[Function]}
      >
        ‚Üê Back
      </span>
    </div>
    <div
      className="banner"
    />
    <Ticket
      currentTicket={null}
      data={null}
    />
  </Modal>
</div>
`;
